index.d.ts

```diff
@@ -0,0 +1,25 @@
+interface CodeReturn<T> {
+  width: number,
+  height: number,
+  data: T
+}
+
+type BufferRet = CodeReturn<Buffer>;
+type UintArrRet = CodeReturn<Buffer>;
+
+export declare function encode(imgData: Buffer, qu: number): BufferRet;
```

Comment:
encode takes `ImageData` which is sorta like `BufferRet | UintArrRet`

Reply:
I may have messed up the typing info for `encode`, I didn't quite understand how `ImageDataLike` and `BufferLike` were both accepted for `encode`, so I did my best guess based on both comments 😅